NPE in requirePluginVersions rule when version location cannot be inferred

### Affected version

3.6.2

### Bug description

```
Caused by: java.lang.NullPointerException: Cannot invoke "String.startsWith(String)" because "modelId" is null
    at org.apache.maven.enforcer.rules.utils.PluginWrapper.isVersionFromDefaultLifecycleBindings (PluginWrapper.java:85)
    at org.apache.maven.enforcer.rules.utils.PluginWrapper.addAll (PluginWrapper.java:53)
    at org.apache.maven.enforcer.rules.RequirePluginVersions.getPlugins (RequirePluginVersions.java:865)
    at org.apache.maven.enforcer.rules.RequirePluginVersions.getAllPluginEntries (RequirePluginVersions.java:817)
    at org.apache.maven.enforcer.rules.RequirePluginVersions.execute (RequirePluginVersions.java:224)
    at org.apache.maven.plugins.enforcer.EnforceMojo.executeRuleNew (EnforceMojo.java:351)
    at org.apache.maven.plugins.enforcer.EnforceMojo.executeRule (EnforceMojo.java:325)
    at org.apache.maven.plugins.enforcer.EnforceMojo.execute (EnforceMojo.java:248)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:569)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

This happens when using the [maven-tiles extension](https://github.com/repaint-io/maven-tiles/) and the version of a maven plugin is defined in a maven tile instead of pom.xml.

--- a/enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/utils/PluginWrapper.java
+++ b/enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/utils/PluginWrapper.java
@@ -81,6 +81,9 @@ public class PluginWrapper {
         }
 
         String modelId = versionLocation.getSource().getModelId();
+        if (modelId == null) {
+            return Optional.empty();
+        }
         return Optional.of(
                 modelId.startsWith("org.apache.maven:maven-core:") && modelId.endsWith(":default-lifecycle-bindings"));
     }
@@ -100,6 +103,9 @@ public class PluginWrapper {
         }
 
         String modelId = versionLocation.getSource().getModelId();
+        if (modelId == null) {
+            return Optional.empty();
+        }
         return Optional.of(
                 modelId.startsWith("org.apache.maven:maven-model-builder:") && modelId.endsWith(":super-pom"));
     }

